docs/CODE-REVIEW-GUIDELINES.md

```diff
@@ -0,0 +1,85 @@
+# Code review guidelines
+
+While solipsism is an interesting philosophical tributary, it’s not a very good way to write long-term maintained software. In general we work in teams at CDS, so we require mandatory code reviews before merging any Pull Request.
+
+Just like all the fun and cool code we love to write often comes down to subjective, individual decisions, the practice of reviewing code is pretty subjective, given that different individuals will do different reviews.
+
+Here are a list of considerations to keep in mind when doing a review: firstly focusing on principles to keep in mind for reviewers, and then leading into technical considerations.
+
+## Principles for reviewers
+
+- [Be nice](https://qz.com/work/625870/after-years-of-intensive-analysis-google-discovers-the-key-to-good-teamwork-is-being-nice/)
+  - If you're not sure how someone wants their code reviewed or how they prefer to receive feedback, talk to them beforehand
+  - read the CDS [Code of Conduct](https://github.com/cds-snc/.github/blob/master/CODE_OF_CONDUCT.md)
+- Point out when something is particularly good or clever
+  - Reviews often focus on negative things so highlight good things as well when you see them
+- Be explicit with feedback
+  - If screenshots or gifs will help make your point, include them
+- Programming is subjective
+  - Different approaches are valid
+  - Ask for clarifications if you need any
+- Communicate which ideas you feel strongly about and those you don't
+- Never review code on weekends
+
+## Technical code review checklist
+
+### Logic
+
+- Does the code work?
+  - Does it perform its intended function?
+  - Is the logic correct?
+  - Is the proposed UI accessible?
+  - Make sure to test the working code yourself
+- Is all the code easily understood?
+  - Which parts are confusing to you and why?
+  - Multiple (nested) if/else blocks are code smells for complex logic
+  - If you are taking a lot of time to understand the code, perhaps it needs comments or refactoring
+- Can the code be simplified?
+  - Remove unused parameters or "generic" code that is only used once
+  - Don't solve problems you don't have
+- Does similar functionality already exist in the codebase?
+  - If so, why isn’t this functionality reused? (Note: it's okay to not reuse things — but it's worth asking the question.)
```

Comment:
I think this touches on an important point - which is that as a review-ee, if someone reviewing your code is asking a question about something, that doesn't mean you've done it wrong. It's perfectly ok to justify your choices and stand by them.

Which is leading me to wonder if maybe a short section on review-ee guidelines might be a good idea?

Reply:
Yeah, not a bad idea. I'll add some.

Reply:
Updated!